source/opaque.d.ts

```diff
@@ -126,6 +126,8 @@ A type returned by `Tagged` can be passed to `Tagged` again, to create a type wi
 
 [Read more about tagged types.](https://medium.com/@KevinBGreene/surviving-the-typescript-ecosystem-branding-and-type-tagging-6cf6e516523d)
 
+A tag is usually a string (and must be a string, number, or symbol), but each application of a tag can also contain an arbitrary type as its metadata. See {@link GetTagMetadata}.
```

Comment:
The docs needs to mention how it affects the type, like assignability.

Reply:
I think the text here should also mention what adding metadata can be useful for.

Reply:
> and people who want invariant behavior can get that using the InvariantOf helper type (e.g., type JsonOf<T> = Tagged<string, "JSON", InvariantOf<T>>) 

Maybe also worth documenting how to get invariant behavior.